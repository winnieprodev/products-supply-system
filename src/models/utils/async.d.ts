export interface Error {
  message: string;
}

export interface Async<T> {
  isLoading: boolean;
  error: Error | undefined;
  data: T | undefined;
}
