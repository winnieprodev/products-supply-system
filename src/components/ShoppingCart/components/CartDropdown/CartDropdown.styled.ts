import styled from "styled-components";

export const Cart = styled.div`
  width: 430px;
  font-size: 13px;
`;

export const Products = styled.div`
  margin: 20px 0 12px;
  overflow: hidden;
`;
