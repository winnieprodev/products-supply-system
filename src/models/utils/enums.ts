export enum WrapperMode {
  full = "full",
  singleLine = "singleLine",
  tiny = "tiny",
}
