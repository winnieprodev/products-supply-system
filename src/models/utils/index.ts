export * from "./async.d";
export * from "./enums";
