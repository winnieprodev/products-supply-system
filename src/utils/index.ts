export * from "./routeUtils";
export * from "./supplierUtils";
export * from "./testUtils";
