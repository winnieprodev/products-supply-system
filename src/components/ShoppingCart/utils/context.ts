import { Async, Cart, ProductCartInfo } from "models";
import React from "react";
import { mockAsync, mockCart } from "utils";

interface ShoppingCartContextProps {
  cartId: Async<number>;
  cart: Async<Cart> | undefined;
  setCart: (cart: Async<Cart> | undefined) => void;
  setProduct: (product: ProductCartInfo) => void;
  removeProduct: (productId: string) => void;
}

export const EMPTY_CART = { products: [] };
const mockFunction = () => {};

export const ShoppingCartContext = React.createContext<ShoppingCartContextProps>(
  {
    cartId: mockAsync(0),
    cart: mockAsync(mockCart(0)),
    setCart: mockFunction,
    setProduct: mockFunction,
    removeProduct: mockFunction,
  },
);
