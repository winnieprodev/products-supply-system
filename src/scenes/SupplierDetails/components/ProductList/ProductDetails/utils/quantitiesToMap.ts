import { Map } from "immutable";
import { CartQuantity, QuantityType } from "models";

export const quantitiesToMap = (quantities: CartQuantity[]) => {
  return Map<QuantityType, number>(
    quantities.map(({ type, amount }) => [type, amount] as const) as Iterable<
      [QuantityType, number]
    >,
  );
};
