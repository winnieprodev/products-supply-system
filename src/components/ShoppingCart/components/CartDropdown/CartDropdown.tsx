import { Button, ProductCartItem } from "components";
import { ShoppingCartContext } from "components/ShoppingCart/utils";
import { useParamsOutsideSwitch } from "hooks";
import { Cart } from "models";
import React, { useContext } from "react";
import { Link } from "react-router-dom";
import ROUTES from "routes";

import * as Styled from "./CartDropdown.styled";

interface OwnProps {
  cart: Cart;
}
const CartDropdown: React.FC<OwnProps> = ({ cart }) => {
  const { outletId = "", supplierId = "", supplierType = "local" } =
    useParamsOutsideSwitch(ROUTES.SUPPLIER_DETAILS) || {};
  const { removeProduct, cartId } = useContext(ShoppingCartContext);

  return (
    <Styled.Cart>
      <Styled.Products>
        {cart.map((product) => (
          <ProductCartItem
            key={product.productId}
            item={product}
            onRemove={() => removeProduct(product.productId)}
            outletId={outletId}
            supplierId={supplierId}
            supplierType={supplierType}
          />
        ))}
      </Styled.Products>
      <Link
        to={ROUTES.CHECKOUT.use({
          outletId,
          supplierType,
          supplierId,
          cartId: cartId.data?.toString() || "",
        })}
      >
        <Button primary fullWidth disabled={cart.length === 0}>
          Checkout
        </Button>
      </Link>
    </Styled.Cart>
  );
};

export default CartDropdown;
