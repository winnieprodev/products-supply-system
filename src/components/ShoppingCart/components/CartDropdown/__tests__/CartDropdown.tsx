import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { mockAsync, mockCart } from "utils";

import CartDropdown from "../CartDropdown";

jest.mock("@apollo/client", () => ({
  gql: (req: string) => req,
  useQuery: () => mockAsync(undefined, true),
}));

describe("CartDropdown", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <BrowserRouter>
          <CartDropdown cart={mockCart(1)} />
        </BrowserRouter>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
