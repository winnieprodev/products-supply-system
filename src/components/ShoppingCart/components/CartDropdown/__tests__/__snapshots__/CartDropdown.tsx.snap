// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropdown Snapshot tests it matches the snapshot 1`] = `
.c2 {
  outline: none;
  cursor: pointer;
  border: 0 none;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  background-color: #F25C05;
  color: #FFFFFF;
}

.c2:hover {
  background-color: #DD5202;
}

.c2:active {
  background-color: #FF7626;
}

.c3 {
  border: 0;
  border-radius: 8px;
  padding: 8px 16px;
  margin: 4px;
  font-weight: 700;
  width: 100%;
  margin: 0;
}

.c0 {
  width: 430px;
  font-size: 13px;
}

.c1 {
  margin: 20px 0 12px;
  overflow: hidden;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
  <a
    href="/checkout//local//0/"
  >
    <button
      class="c2 c3"
    >
      Checkout
    </button>
  </a>
</div>
`;
